lib/board.rb

```diff
@@ -0,0 +1,60 @@
+class Board 
+  attr_accessor :cells 
+  
+  
+  def initialize
+    @cells = Array.new(9," ")
+  end
+
+  def reset!
+    self.cells =[" ", " ", " ", " ", " ", " ", " ", " ", " "]
+  end
+
+  def display
+    puts " #{@cells[0]} | #{@cells[1]} | #{@cells[2]} "
+    puts "-----------"
+    puts " #{@cells[3]} | #{@cells[4]} | #{@cells[5]} "
+    puts "-----------"
+    puts " #{@cells[6]} | #{@cells[7]} | #{@cells[8]} "
+    ## line spacing
+    puts ""
+    puts""
+  end
+
+
+  def position(position)
+     @cells[position.to_i-1]
+  end
+
+  def full?
+    @cells.all?{|piece|piece == "X" || piece == "O"}
+  end
+
+  def turn_count
+    x=0
+    @cells.each do |piece|
+      if piece == "X" || piece == "O"
+        x+=1
+      end
+    end
+    x
```

Comment:
Use the highest level and most expressive iterator possible to solve a problem. [`count`](http://ruby-doc.org/core-2.2.0/Array.html#method-i-count) on the cells is perfect and the rest of this method is not needed. Entire body could read:

```ruby
def turn_count
  @cells.count {|piece| piece == "X" || piece == "O"}
end
```